File: back-end/src/main/java/pl/straburzynski/campfiresongs/model/SessionDto.java
package pl.straburzynski.campfiresongs.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.UUID;

@Builder
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class SessionDto {

    private UUID id;

    private SongDto song;

    private String name;

    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private String password;

    private boolean temporary;

    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private String songData;

}

File: back-end-kotlin/src/main/kotlin/pl/straburzynski/campfiresongs/session/model/SessionDto.kt
package pl.straburzynski.campfiresongs.session.model

import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonProperty.Access.WRITE_ONLY
import pl.straburzynski.campfiresongs.song.model.SongDto
import java.util.UUID

data class SessionDto(
    val id: UUID?,
    val song: SongDto?,
    val name: String,
    @JsonProperty(access = WRITE_ONLY)
    val password: String?,
    @JsonProperty(access = WRITE_ONLY)
    val temporary: Boolean,
    val songData: String?
)
